File: codingame/easy/power-of-thor-episode-1.py
a,b,c,d=map(int,input().split())
e,f=c,d
while 1:
 _=input();h=""
 if f!=b:h,f=((h+"S",f+1)if f<b else(h+"N",f-1))
 if e!=a:h,e=((h+"E",e+1)if e<a else(h+"W",e-1))
 print(h)

File: codingame/easy/power-of-thor-episode-1.js

var inputs = readline().split(' ');
lightX = parseInt(inputs[0]);
lightY = parseInt(inputs[1]);
initialTx = parseInt(inputs[2]);
initialTy = parseInt(inputs[3]);

while (true) {
    const remainingTurns = parseInt(readline());
    
    res = ""
    if (initialTy > lightY) {
        res += "N"
        initialTy -= 1}
    else if (initialTy < lightY) {
        res += "S"
        initialTy += 1}
    if (initialTx > lightX) {
        res += "W"
        initialTx -= 1}
    else if (initialTx < lightX) {
        res += "E"
        initialTx += 1
    }
    
    print(res)
}
    